### PR TITLE
feat(feedback): show issue details' TraceDataSection in feedback details

### LINKED ISSUE #68481: [User Feedback] - Issue Trace Timeline
tl;dr: Add the new trace timeline to User Feedback details

A new component was built for the Issue Details page called [Issue Trace Timeline](https://sentry.io/changelog/issues-trace-timeline/). Previously known as Trace Navigator

image

With the JS SDK v8, each new page load or navigation will create a new trace. That means that if a user submits a User Feedback through the widget, that feedback will be using the same trace-id as other relevant data during that last page load. Including client side errors, as well as calls to the backend that might have be the culprit of that error.

With the JS SDK v7 though, there is a global trace id that wasn't used during page load requests that resulted in HTTP calls to the backend. This means that User Feedbacks from the widget are not likely to be connected to any other telemetry.

Additionally, feedback received from the legacy API (that only takes name/email/comments) doesn't have a trace id. They are connected to an error though (through `event_id`) so we potentially leverage that trace-id. We can do so by coping the `trace_id` from the [event linked in the shim code](https://github.com/getsentry/sentry/blob/1d5adc7369f53bf2e9396b1aba8a0e21f67cd5f1/src/sentry/feedback/usecases/create_feedback.py#L290-L302).

To verify the data, we can use `?dataset=issuePlatform` on Discover. For example [this query](https://sentry.sentry.io/discover/homepage/?dataset=issuePlatform&field=user.display&field=timestamp&field=trace&name=All+Events&project=11276&query=title%3A%22User+Feedback%22&sort=-timestamp&statsPeriod=14d&yAxis=count%28%29). Once v8 is installed in Sentry we'll be able to confirm what data would be visible in the component.

Also note the component might need to be adjusted to show User Feedback. It was designed for Issue Details so it shows other error events only (no spans).

<details>
<summary>Additional context</summary>

## Spans on trace navigator

image

## Trace-id behavior in JS SDK v7 vs v8

image

image
image

</details>

### PR DESCRIPTION
**Pull request by aliu39**
Fixes https://github.com/getsentry/sentry/issues/68481

For the screenshots, I used dev-ui to override the feedback's trace ID. Before merging, we need to test this by finding/creating some real-world, prod examples.

1. [Trace w/26 issues:](https://sentry.sentry.io/performance/trace/e0078a14f2e34eab85af36dc36fbfc1c/?eventId=e7f2609d56af4b478dda76748f44ebf6&feedbackSlug=javascript%3A5721692549&groupId=5721692549&pageEnd=2024-08-17T10%3A29%3A42.585&pageStart=2024-08-16T10%3A29%3A42.585&query=&referrer=feedback_list_page&source=issue_details&timestamp=1723847311)

trace-ex1

_You can try interacting with the same timeline in [this issue](https://sentry.sentry.io/issues/5545076208/?query=is%3Aunresolved+issue.priority%3A%5Bhigh%2C+medium%5D+invalidsearchquery&referrer=issue-stream&statsPeriod=14d&stream_index=2)_.

2. [Trace w/1 other issue:](https://sentry.sentry.io/performance/trace/a867ab58e5014df5a8e3168549d3fcdc/?eventId=e7f2609d56af4b478dda76748f44ebf6&feedbackSlug=javascript%3A5721692549&groupId=5721692549&pageEnd=2024-08-17T10%3A29%3A42.585&pageStart=2024-08-16T10%3A29%3A42.585&referrer=feedback_list_page&source=issue_details&timestamp=1723847311)

trace-ex2

3. If missing trace, or trace has 0 other issues, _the section is not rendered at all_

### CODE REVIEW

File: static/app/components/feedback/feedbackItem/feedbackItem.tsx [hunk 1]
```diff
@@ -122,6 +132,34 @@ export default function FeedbackItem({feedbackItem, eventData, tags}: Props) {
   );
 }
 
+function TraceDataSection({
```
Comment (michellewzhang) [inline static/app/components/feedback/feedbackItem/feedbackItem.tsx:135]: can we move this to a new file? similar to how the other sections are structured

**Review by michellewzhang — APPROVED** [review]
before we merge, could we find some real examples to test this component on? i bet @bruno-garcia could dig up some from nugettrends or elsewhere

**Comment by aliu39** [comment]
> before we merge, could we find some real examples to test this component on? i bet @bruno-garcia could dig up some from nugettrends or elsewhere

Definitely! To be clear, we need 2+ unique issues in the same trace as the feedback. Excluding the feedback itself

**Comment by aliu39** [comment]
Here's a good prod example: https://sentry.dev.getsentry.net:7999/feedback/?feedbackSlug=javascript%3A5764397098&project=11276
When first created this had a timeline, but since the errors were the same, they were grouped to 1 issue.

https://sentry.dev.getsentry.net:7999/feedback/?feedbackSlug=javascript%3A5736701211
Example with both a trace-related issue and a crash report. Note the trace link (/performance) is different from the crash report link (/issues/:groupId)

**Comment by michellewzhang** [comment]
looks good! there's some timeline examples on nugettrends that look good to me

File: static/app/components/feedback/feedbackItem/feedbackItem.tsx (hunk 1 above)
Comment (aliu39) [reply static/app/components/feedback/feedbackItem/feedbackItem.tsx:135]: hm I feel like it's small enough to keep here, it's basically a wrapper around Issue's TraceDataSection

**Comment by bruno-garcia** [comment]
Here's a feedback with a trace:

[Feedback page](https://nugettrends.sentry.io/feedback/?feedbackSlug=spa%3A5721322639&project=1266321&referrer=feedback_list_page&statsPeriod=90d)
[Feedback in issue detail](https://nugettrends.sentry.io/issues/5721322639/?project=1266321&referrer=issue-list&statsPeriod=90d)
[The crash report event linked to the feedback for reference] (https://nugettrends.sentry.io/issues/5574616067/events/a17b6754ff264779a6f7ba96fc6616e7/?project=1266321&referrer=event-or-group-header&statsPeriod=90d)

**Comment by aliu39** [comment]
Reopened at https://github.com/getsentry/sentry/pull/76858

**Pull request closed without merging**